nswbuff.el

```diff
@@ -336,6 +336,12 @@ This map becomes active whenever ‘nswbuff-switch-to-next-buffer’ or
 bind functions for buffer handling which then become available
 during buffer switching.")
 
+;; Make sure status buffer is reset when window layout is changed.
+(add-variable-watcher
+ 'nswbuff-status-window-layout
+ (lambda (_oldval _newval _operation _where)
+   (setq nswbuff-status-buffer nil)))
+
```

Comment:
Are you sure a variable watcher is necessary, given the changes below?

Reply:
Yes, in my test if you do the following sequence you have a minibuffer cleanup issue again.

- 0/ (setq nswbuff-status-window-layout  nil) ;; startup, 
- 1/ (setq nswbuff-status-window-layout 'minibuffer);; nswbuff-status-buffer points to minibuffer
- 2/ (setq nswbuff-status-window-layout nil) ;; nswbuff-status-buffer still points to minibuffer

So after step 2 you will keep writing to `" *nswbuff*"` and `" *Minibuf-0*"` which is why I added this watcher.

Reply:
I think this is required because `kill-buffer` won't kill the minibuffer (I think you can't kill the minibuffer in Emacs)

Reply:
Ah, I didn't think that far ahead... I must admit I simply would have put a note in the documentation saying that if one changes `nswbuff-status-window-layout`, one should restart Emacs. :smiley: Your solution is definitely better, though.